Code/Framework/AzToolsFramework/AzToolsFramework/FocusMode/FocusModeNotificationBus.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/EBus/EBus.h>
+
+namespace AzToolsFramework
+{
+    class FocusModeNotifications
+        : public AZ::EBusTraits
+    {
+    public:
+        virtual ~FocusModeNotifications() = default;
+
+        virtual void OnEditorFocusChanged(AZ::EntityId entityId) = 0;
+    };
+
+    using FocusModeNotificationBus = AZ::EBus<FocusModeNotifications>;
+
+} // namespace AzToolsFramework
```

Comment:
I don't think we should add this bus, I think we should use the central system @jonawals has been working on correct? This seems redundant and something that is duplicating the events being sent

Reply:
Actually I think I realize this is a slightly different to the overall editor states (as this is for changing entities) - API comments and some tweaks to the naming might help

Reply:
This should likely be identified by viewport id too, as this will break if we have multiple viewports open in future

Reply:
After discussing this online, I am changing this to be addressed by AzFramework::EntityContextId